editor/header/tools/style.scss

```diff
@@ -75,3 +75,34 @@
 		}
 	}
 }
+
+.editor-tools__publish-button.is-saving,
+.editor-tools__publish-button.is-saving:disabled {
+	position: relative;
+
+	// These styles overrides the disabled state with the button primary styles
+	background: none !important;
```

Comment:
Unfortunately we can't. These are the styles of WP and they are marked as `important` :(